### PR TITLE
Split "gfx anims" into "pic anims" and "sprite anims"

### LINKED ISSUE #876: Separate "gfx anims" macros into "pic anims", "battle anims", and "sprite anims"
There are some similarities between these two systems, but enough differences that they shouldn't be sharing macros.

Pic anims (for animated Pokemon) use `frame`s with durations, plus `setrepeat` and `dorepeat` for loops, with `endanim` (just the usual `db -1`) to end them.

Battle and sprite anims (for OAM sequences, in the battle engine and the general-purpose sprite engine) use `frame`s with durations but also X/Y flip flags (shifted into the high two bits so long durations can go in the low six), plus `dorestart`, `dowait`, and `delanim` for further controls, with `endanim` to end them.

These three systems mostly do not use the same code: what they share is a notion of `frame`s being shown over durations. Having separate macros and documentation would make their differences more clear.

(As a separate issue, we should look further into the similarities between battle and sprite anims, and settle on more consistent terminology for them. Compare the data/battle_anims/ and data/sprite_anims/ files to see how they correspond, as well as battle_anim_constants.asm vs sprite_anim_constants.asm.)

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #876

### CODE REVIEW

**Comment by Rangi42** [comment]
@aaaaaa123456789 @mid-kid I introduced the `spriteanimoam` macro partly to correspond with `battleanimoam`, but also because I seem to recall some project that changed the data format and ended up introducing such a macro anyway (although possibly not for this table, but something else sprite/battle anim related). It doesn't look like pokecrystal16 or @SoupPotato's sprite+move anim project were it. Does this sound familiar?

**Comment by coltongit** [comment]
Looks fairly straightforward from what I can tell of it. LGTM!

File: data/sprite_anims/oam.asm [hunk 1]
```diff
@@ -1,147 +1,152 @@
+MACRO spriteanimoam
+; vtile offset, data pointer
+	db \1
+	dw \2
+ENDM
```
[the remark this answers is not in the document]
Comment (Rangi42) [reply data/sprite_anims/oam.asm:5]: It follow the usual pattern for macros specific to some data table: define them right above the table.

**Review by aaaaaa123456789 — APPROVED** [review]
Looks good to me.

**Comment by mid-kid** [comment]
I will always prefer splitting the object files up such that only the oam animation macros are visible in the framesets.asm files and the sprite animation macros are only visible in the pokemon sprite anim files. Having to make a distinction as inane as `doframe` vs `frame` and `stopanim` vs `endanim` will never not bother me, as it's impossible to tell each apart in a vacuum.

If you really need both subsystem's macros to be visible simultaneously, use names relevant to the subsystem instead. e.g. `oamframe` or `pokeframe`, or `endoamanim` or `oamend` or any combination that makes it obvious which macro serves for `oam` and which one for `poke`.

**Comment by aaaaaa123456789** [comment]
While I agree that respelling keywords is a bad solution to _everything_ (see: `callfar` vs. `farcall`), I think this is an improvement over the previous situation.

**Comment by mid-kid** [comment]
The difference with `callfar` vs `farcall` and this is that we're trying to clarify the incompatibility between the two values which are otherwise identical, because they belong to separate subsystems and may be modified to mean different things. Both farcall variants are literally identical for every purpose.

**Comment by mid-kid** [comment]
If I had to suggest a set of macros for oam anims on a whim, I'd go with prefixing everything for consistency:
* doframe -> oamframe
* stopanim -> oamend
* dorestart -> oamrestart
* dowait -> oamwait
* delanim -> oamdel (what does this one even mean?)

This preserves compatibility for poke sprite animations, and we were breaking compat for oam macros anyway.

**Comment by vulcandth** [comment]
> * delanim -> oamdel (what does this one even mean?)


According to a deleted comment it removes an object from the screen as opposed to just stopping it.

**Comment by aaaaaa123456789** [comment]
`oamdestroy`, perhaps?

**Comment by mid-kid** [comment]
`oamobliterate` :stuck_out_tongue_closed_eyes: 

Either `oamdel` or `oamdestroy` are ok, just needs a comment or documentation somewhere. I'd almost expect `oamend` to destroy the oam object.

**Comment by coltongit** [comment]
> If I had to suggest a set of macros for oam anims on a whim, I'd go with prefixing everything for consistency:
> 
>     * doframe -> oamframe
> 
>     * stopanim -> oamend
> 
>     * dorestart -> oamrestart
> 
>     * dowait -> oamwait
> 
>     * delanim -> oamdel (what does this one even mean?)
> 
> 
> This preserves compatibility for poke sprite animations, and we were breaking compat for oam macros anyway. They're completely distinguishable, look wrong in pokemon animations, and it's hard to accidentally use `dorepeat` instead of `dorestart`.

Most of these look good, `delanim` could be `oamdelanim` as not to lose what it's purpose is supposed to be

**Comment by Rangi42** [comment]
@mid-kid Thanks, I've made those updates.

**Pull request merged**